codec/codec.go

```diff
@@ -0,0 +1,46 @@
+package codec
```

Comment:
You could grab the codec's name (this is the only bit that is needed in transport, right?) and stuff that in the `transport.Stream` struct, no?